cdci_data_analysis/analysis/parameters.py

```diff
@@ -701,7 +762,7 @@ def value(self, v):
                 self.check_value(v, par_format=self.par_format, name=self.name)
             if self._allowed_values is not None:
                 if v not in self._allowed_values:
-                    raise RuntimeError(f'value {v} not allowed, allowed= {self._allowed_values}')
+                    raise RuntimeError(f'Parameter {self.name} wrong value {v}: not in allowed {self._allowed_values}')
```

Comment:
just wondering, why is it `RuntimeError` and not something like [`RequestNotUnderstood`](https://github.com/oda-hub/dispatcher-app/blob/master/cdci_data_analysis/analysis/exceptions.py#L37)?

`RuntimeError` can be many things, and we unsurprisingly had issues when internal errors are sent to users unnecessarily. Could be better to distinguish right here that this is specifically an issue with request parameters.

Reply:
It was RuntimeError before. And in most similar checks of this module it is.
In check_bounds I've made it to be ValueError though. 
The error message describes that the issue is with parameter value.
So what do you propose?

Reply:
I propose to change to more specific exception class.

Also, just for clarify, to check how `RuntimeError` is caught so it converts to 400 error - if this happens there may be false 400 messages.

Reply:
And in the integral plugin, the realization of bounds checking actually raises RequestNotUnderstood. I can change all exceptions here to it. But what "it is not user error" means in the exception comment?

Reply:
it's just to be polite. some people just react better when we are saying "we can not understand this, you probably meant something else" instead of "you are wrong because you made an error"

Reply:
Ok, got it

Concerning exception class, I see now that it definitely must be RequestNotUnderstood. It's a subclass of APIError and it's treated specifically to not clutter the output with unnecessary info. Will change it throughout the module

Reply:
I changed the exception class for those occurrences where I am sure the wrong value could come from user input. Not sure about others so left them as-is